src/components/data/details/PermissionsPanel.js

```diff
@@ -0,0 +1,250 @@
+                        </Grid>
+                        <Grid item sm={6} xs={6}>
+                            <Skeleton variant="text" />
+                        </Grid>
+                    </Fragment>
+                ))}
+            </Grid>
+        );
+    }
+
+    const groupedPermissions = groupBy(
+        sortPerms(permissions),
+        (permission) => permission.permission
+    );
+
+    const getAvatarClass = (permissionValue) => {
+        if (permissionValue === Permissions.OWN) {
+            return classes.avatarOwn;
+        }
+        if (permissionValue === Permissions.WRITE) {
+            return classes.avatarWrite;
+        }
+        return classes.avatarRead;
+    };
+
+    return (
+        <>
+            <Typography id={build(baseId, ids.SELF_PERMISSION)}>
+                {getMessage("selfPermission", {
+                    values: { permission: selfPermission },
+                })}
+            </Typography>
+
+            <List
+                className={classes.list}
+                id={build(baseId, ids.PERMISSION_LIST)}
+            >
+                {Object.values(Permissions).map((permissionValue, index) => (
+                    <Fragment key={index}>
+                        <ListSubheader>
+                            {permissionValue} (
+                            {groupedPermissions[permissionValue]?.length || 0})
+                        </ListSubheader>
+                        {groupedPermissions &&
+                            groupedPermissions[
+                                permissionValue
+                            ]?.map((permission, index) => (
+                                <Identity
+                                    id={build(
+                                        baseId,
+                                        ids.PERMISSION_LIST,
+                                        permission.user
+                                    )}
+                                    key={index}
+                                    primaryText={permission.name}
+                                    secondaryText={
+                                        permission.institution
+                                            ? permission.institution
+                                            : permission.description
+                                    }
+                                    secondaryAction={
+                                        <>
+                                            {permissionLoadingId ===
+                                            permission.user ? (
+                                                <CircularProgress
+                                                    color="inherit"
+                                                    size={20}
+                                                />
+                                            ) : (
+                                                <PermissionSelector
```

Comment:
Nice! I really like this concept ✨ 

It seems this only supports 1 `permissionLoadingId` at a time, however. I can see the permission service taking a long time to respond, and in the meantime the user might want to update another user's permissions (or more than 1 in a row).

Can slow responses be simulated in storybook with `fetchMock`?

Reply:
That's an excellent point.  I'll refactor this with multiple updates in mind 👍